runtime/src/accounts_db.rs

```diff
@@ -1475,13 +1650,34 @@ impl AccountsDB {
                         .map(|i| i.accounts.capacity())
                         .sum::<u64>();
                 }
-                all_removed_slot_storages.push(slot_removed_storages);
+                all_removed_slot_storages.push(slot_removed_storages.clone());
             }
         }
         remove_storages_elapsed.stop();
 
         let num_slots_removed = all_removed_slot_storages.len();
 
+        let mut recycled_count = 0;
+        let mut recycle_stores = self.recycle_stores.write().unwrap();
+        for slot_entries in &all_removed_slot_storages {
+            let entry = slot_entries.read().unwrap();
+            for (_slot, stores) in entry.iter() {
+                if recycle_stores.len() > MAX_RECYCLE_STORES {
+                    break;
+                }
+                recycle_stores.push(stores.clone());
+                recycled_count += 1;
+            }
+            if recycle_stores.len() > MAX_RECYCLE_STORES {
+                let dropped_count = total_removed_storage_entries - recycled_count;
+                self.stats
+                    .dropped_stores
+                    .fetch_add(dropped_count as u64, Ordering::Relaxed);
+                break;
+            }
+        }
+        drop(recycle_stores);
```

Comment:
I think it would be cleaner to move this block into a `recycle_slot_stores(&self, removed_slot_storages: &[SlotStores])` function. 

That way we can move the outer break into the inner `break` and `return` instead, and we don't need to explicitly drop the `recycle_stores` lock

Reply:
Added the helper.